test/unit/conditions/base/contract.test.ts

```diff
@@ -0,0 +1,154 @@
+import { ContractCondition } from '../../../../src/conditions/base';
+import { testContractConditionObj } from '../../testVariables';
+
+describe('validation', () => {
+  it('accepts on a valid schema', () => {
+    const contract = new ContractCondition(testContractConditionObj);
+    expect(contract.toObj()).toEqual({
+      ...testContractConditionObj,
+    });
+  });
+
+  it('rejects an invalid schema', () => {
+    const badContractCondition = {
+      ...testContractConditionObj,
+      // Intentionally removing `contractAddress`
+      contractAddress: undefined,
+    };
+    const badEvm = new ContractCondition(badContractCondition);
+    expect(() => badEvm.toObj()).toThrow(
+      'Invalid condition: "contractAddress" is required'
+    );
+
+    const { error } = badEvm.validate(badContractCondition);
+    expect(error?.message).toEqual('"contractAddress" is required');
+  });
+});
+
+describe('accepts either standardContractType or functionAbi but not both or none', () => {
+  const standardContractType = 'ERC20';
+  const functionAbi = {
+    inputs: [
+      {
+        name: '_owner',
+        type: 'address',
+      },
+    ],
+    name: 'balanceOf',
+    outputs: [
+      {
+        name: 'balance',
+        type: 'uint256',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  };
+
+  it('accepts standardContractType', () => {
+    const conditionObj = {
+      ...testContractConditionObj,
+      standardContractType,
+      functionAbi: undefined,
+    };
+    const contractCondition = new ContractCondition(conditionObj);
+    expect(contractCondition.toObj()).toEqual({
+      ...conditionObj,
+    });
+  });
+
+  it('accepts functionAbi', () => {
+    const conditionObj = {
+      ...testContractConditionObj,
+      functionAbi,
+      standardContractType: undefined,
+    };
+    const contractCondition = new ContractCondition(conditionObj);
+    expect(contractCondition.toObj()).toEqual({
+      ...conditionObj,
+    });
+  });
+
+  it('rejects both', () => {
+    const conditionObj = {
+      ...testContractConditionObj,
+      standardContractType,
+      functionAbi,
+    };
+    const contractCondition = new ContractCondition(conditionObj);
+    expect(() => contractCondition.toObj()).toThrow(
+      '"value" contains a conflict between exclusive peers [standardContractType, functionAbi]'
+    );
+  });
+
+  it('rejects none', () => {
+    const conditionObj = {
+      ...testContractConditionObj,
+      standardContractType: undefined,
+      functionAbi: undefined,
+    };
+    const contractCondition = new ContractCondition(conditionObj);
+    expect(() => contractCondition.toObj()).toThrow(
+      '"value" must contain at least one of [standardContractType, functionAbi]'
+    );
+  });
+});
+
+// TODO(#124)
+// it('accepts custom parameters in function abi methods', async () => {
+//     throw new Error('Not implemented');
+// });
+
+// TODO(#124)
```

Comment:
Hmm those two should be fixed by now

Reply:
This was the file that was removed in `tdec-epic` by a different PR. I re-added it and renamed it after a rebase. It is uncommented, and slightly modified now.